Update geo-utils.R

tomsing1 opened the pull request:
Playing with additional SRA projects, I had to correct a few assumptions:
- The `cell_type` attribute may not exist, so I cannot hard-code its name.
- Biosample annotation rows need to be made distinct to enable `tidyr::spread` to work.

tomsing1 committed:
Update geo-utils.R

tomsing1 committed:
map_df instead of bind_rows

lianos reviewed (approved):
Just remove the `@importFrom` on the `dplyr` and `purrr` packages, then go ahead and just merge this puppy.

tomsing1 committed:
Removed dplyr and purrr imports

tomsing1 committed:
Added retrieve_sra_metadata helper function

The `retrieve_sra_metadata` function retrieves additional metadata provided by either the EBI or NCBI as tsv/csv files, including URLs for downloading the raw data.
